File: 2_Ruby_Examples/1_ConferenceAttendees_SRP/price_calculator.rb
require 'attendee'
require 'attendee_type'

class PriceCalculator
	
	REGULAR_PRICE = 1000
	
	def calculatePrice attendee
		if attendee.type == AttendeeType::SPEAKER
			REGULAR_PRICE / 2
		elsif attendee.type == AttendeeType::ORGANIZER
			0
		elsif attendee.type == AttendeeType::PARTICIPANT
			REGULAR_PRICE
		end
	end
	
end

File: 4_Clojure_Examples/1_ConferenceAttendees_SRP/price_calculator.clj
(ns price_calculator)

(def REGULAR-PRICE 1000)

(defn calculate-price [attendee]
	(cond (= :participant (:type attendee)) REGULAR-PRICE
			  (= :speaker (:type attendee)) (/ REGULAR-PRICE 2)
			  (= :organizer (:type attendee)) 0))
